### PR TITLE
[IMPROVEMENT] Open admin panel fullscreen on iPad

### LINKED ISSUE #1137: [IMPROVEMENT] Change iPad implementation to open Admin in a full screen controller

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/ios

Also changed the "Admin" button to be a cell in another section.

TODO:

- [x] move admin button to side bar and open fullscreen

Closes #1137

### CODE REVIEW

**Comment by rafaelks** [comment]
I don't think this transition is good... what do you think about changing our admin to open in a modal instead of pushing on the navigation stack?

**Comment by cardoso** [comment]
@rafaelks UIKit doesn't allow modal over modal, but i could do a workaround.

File: Rocket.Chat/de.lproj/Localizable.strings [hunk 1]
```diff
@@ -83,6 +83,7 @@
 "user_menu.away" = "Abwesend";
 "user_menu.busy" = "Beschäftigt";
 "user_menu.invisible" = "Unsichtbar";
+"user_menu.admin" = "Administration"; //TODO
```
Comment (soundstorm) [inline Rocket.Chat/de.lproj/Localizable.strings]: No need for //TODO ;)

File: Rocket.Chat/Controllers/Settings/SettingsViewController.swift [hunk 2]
```diff
@@ -33,15 +33,17 @@ final class SettingsViewController: UITableViewController {
...
-
         title = viewModel.title
+    }
 
-        if viewModel.canViewAdminPanel {
-            let buttonAdmin = UIBarButtonItem(title: "Admin", style: .plain, target: self, action: #selector(buttonAdminDidPressed(_:)))
-            navigationItem.rightBarButtonItem = buttonAdmin
-        }
+    override func viewWillAppear(_ animated: Bool) {
+        navigationController?.fullScreen = false
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Settings/SettingsViewController.swift]: Is this still required?

File: Rocket.Chat/Controllers/Settings/SettingsViewController.swift [hunk 3]
```diff
@@ -50,17 +52,18 @@ final class SettingsViewController: UITableViewController {
...
             let adminURL = URL(string: "\(baseURL)/admin/info?layout=embedded")
-        else {
-            return
+            else {
+                return
         }
 
         if let controller = WebViewControllerEmbedded.instantiateFromNib() {
             controller.url = adminURL
+            navigationController?.fullScreen = true
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Settings/SettingsViewController.swift]: Required?

File: Rocket.Chat/Controllers/Settings/SettingsViewController.swift [hunk 4]
```diff
@@ -50,17 +52,18 @@ final class SettingsViewController: UITableViewController {
         }
     }
 
-    @objc func buttonAdminDidPressed(_ sender: Any) {
+    func openAdminPanel() {
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Settings/SettingsViewController.swift]: Required?

File: Rocket.Chat/Controllers/Settings/SettingsViewController.swift [hunk 5]
```diff
@@ -103,21 +106,23 @@ final class SettingsViewController: UITableViewController {
             }
         }
 
-        #if DEBUG || BETA
-            if indexPath.section == 1, indexPath.row == 0 {
-                FLEXManager.shared().showExplorer()
-            }
-        #endif
+        if indexPath.section == 1, indexPath.row == 0 {
+            openAdminPanel()
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Settings/SettingsViewController.swift]: I believe we don't have it anymore here, right?

File: Rocket.Chat/SettingsViewModel.swift [hunk 6]
```diff
@@ -66,6 +61,24 @@ final class SettingsViewModel {
...
+        return 2
+    }
+
+    internal func numberOfRowsInSection(_ section: Int) -> Int {
+        switch section {
+        case 0: return 3
+        case 1: return (canOpenFLEX ? 1 : 0)
+        default: return 0
+        }
+    }
```
Comment (rafaelks) [inline Rocket.Chat/SettingsViewModel.swift]: Is this code required?

File: Rocket.Chat/Views/Subscriptions/SubscriptionUserStatusView.swift [hunk 7]
```diff
@@ -115,4 +145,20 @@ final class SubscriptionUserStatusView: UIView {
...
+    func openAdminPanel() {
+        guard
+            let auth = AuthManager.isAuthenticated(),
+            let baseURL = auth.settings?.siteURL,
+            let adminURL = URL(string: "\(baseURL)/admin/info?layout=embedded")
+        else {
+            return
+        }
+
+        if let controller = WebViewControllerEmbedded.instantiateFromNib() {
```
Comment (rafaelks) [inline Rocket.Chat/Views/Subscriptions/SubscriptionUserStatusView.swift]: I think you can add it to a BaseNavigationController, top is staying white now... can you take a look?

**Comment by cardoso** [comment]
Oops! Made the changes @rafaelks

**Pull request merged**